frontstage/views/surveys/add_survey.py

```diff
@@ -27,25 +27,33 @@ def add_survey(_):
         try:
             iac = iac_controller.get_iac_from_enrolment(enrolment_code)
             if iac is None:
-                logger.info("Enrolment code not found")
+                logger.info("Enrolment code not found when attempting to add survey", enrolment_code=enrolment_code)
                 template_data = {"error": {"type": "failed"}}
                 return render_template("surveys/surveys-add.html", form=form, data=template_data), 200
             if not iac["active"]:
-                logger.info("Enrolment code not active")
+                logger.info("Enrolment code not active when attempting to add survey", enrolment_code=enrolment_code)
                 template_data = {"error": {"type": "failed"}}
                 return render_template("surveys/surveys-add.html", form=form, data=template_data)
         except ApiError as exc:
             if exc.status_code == 400:
-                logger.info("Enrolment code already used", status_code=exc.status_code)
+                logger.info(
+                    "Enrolment code already used when attempting to add survey",
+                    status_code=exc.status_code,
+                    enrolment_code=enrolment_code,
+                )
                 template_data = {"error": {"type": "failed"}}
                 return render_template("surveys/surveys-add.html", form=form, data=template_data)
             else:
-                logger.error("Failed to submit enrolment code", status_code=exc.status_code)
+                logger.error(
+                    "Failed to submit enrolment code when attempting to add survey",
+                    status_code=exc.status_code,
+                    enrolment_code=enrolment_code,
+                )
                 raise
 
         cryptographer = Cryptographer()
         encrypted_enrolment_code = cryptographer.encrypt(enrolment_code.encode()).decode()
-        logger.info("Successful enrolment code submitted")
+        logger.info("Successful enrolment code submitted when attempting to add survey", enrolment_code=enrolment_code)
```

Comment:
I don't think its successfully submitted
Can you add a log line to suggest that all validation has been completed and now we will decrypt 
another on that the decoding was done sucessfully
and once we actually save the info in db then we suggest it has been sumitted.